web/ASC.Web.Common/src/components/MediaViewer/sub-components/video-viewer.js

```diff
@@ -0,0 +1,473 @@
+      if (this.props.url !== prevProps.url) {
+        newUrl = this.props.url
+      }
+      if (this.props.playing !== prevProps.playing) {
+        newPlaying = this.props.playing
+      }
+      this.setState(
+        {
+          url: newUrl,
+          playing: newPlaying
+        }
+      );
+    }
+  }
+
+  handlePlayPause = () => {
+    this.setState({ playing: !this.state.playing })
+  }
+
+  handleStop = () => {
+    this.setState({ url: null, playing: false })
+  }
+
+  handleVolumeChange = e => {
+    this.setState({
+      volume: parseFloat(e.target.value),
+      muted: false
+    })
+  }
+
+  handleToggleMuted = () => {
+    this.setState({ muted: !this.state.muted })
+  }
+
+  handlePlay = () => {
+    this.setState({ playing: true })
+  }
+
+  handleEnablePIP = () => {
+    this.setState({ pip: true })
+  }
+
+  handleDisablePIP = () => {
+    this.setState({ pip: false })
+  }
+
+  handlePause = () => {
+    this.setState({ playing: false })
+  }
+
+  handleSeekMouseDown = e => {
+    this.setState({ seeking: true })
+  }
+
+  handleSeekChange = e => {
+    this.setState({ played: parseFloat(e.target.value) })
+  }
+
+  handleSeekMouseUp = e => {
+    if (!isNaN(parseFloat(e.target.value))) {
+      this.setState({ seeking: false });
+      this.player.seekTo(parseFloat(e.target.value));
+    }
+  }
+
+  handleProgress = state => {
+    if (!this.state.seeking) {
+      this.setState(state)
+    }
+  }
+
+  handleEnded = () => {
+    this.setState({ playing: this.state.loop })
+  }
+
+  handleDuration = (duration) => {
+    this.setState({ duration })
+  }
+
+  handleClickFullscreen = () => {
+    screenfull.request(findDOMNode(this.player))
+  }
+
+  renderLoadButton = (url, label) => {
+    return (
+      <button onClick={() => this.load(url)}>
+        {label}
+      </button>
+    )
+  }
+
+  ref = player => {
+    this.player = player
+  }
+
+  resizePlayer = (videoSize, screenSize) => {
+    var ratio = videoSize.h / videoSize.w;
+
+    if (videoSize.h > screenSize.h) {
+      videoSize.h = screenSize.h;
+      videoSize.w = videoSize.h / ratio;
+    }
+    if (videoSize.w > screenSize.w) {
+      videoSize.w = screenSize.w;
+      videoSize.h = videoSize.w * ratio;
+    }
+
+    return {
+      width: videoSize.w,
+      height: videoSize.h
+    };
+
+  };
+
+  render() {
+    const { url, playing, controls, light, volume, muted, loop, played, loaded, duration, playbackRate, pip } = this.state
+    const parentOffset = this.props.getOffset() || 0;
+    var screenSize = {
+      w: window.innerWidth,
+      h: window.innerHeight
+    };
+    screenSize.h -= parentOffset + controlsHeight;
+
+    let width = screenSize.w;
+    let height = screenSize.h;
+
+    var centerAreaOx = screenSize.w / 2 + document.documentElement.scrollLeft;
+    var centerAreaOy = screenSize.h / 2 + document.documentElement.scrollTop;
+
+    if (document.getElementsByTagName('video')[0]) {
```

Comment:
Move "document.getElementsByTagName('video')[0]" to single const and use it for videoWidth/videoHeight